### DISCUSSION TITLE
Download historical video

### DISCUSSION THREAD

**Discussion by Joe-Wulf** [dgreif/ring#691] [Q&A]
I'm experienced and literate with Linux/RedHat/CentOS; bash scripting, OS installation/security/management and the like.  Not familiar nor experienced with perl/python, Java, Typescript libraries, Git/GitLab, using API's, etc...   I'm willing to dig in and learn and contribute back from my failures and lessons; though I'm initially limited by time.

I'm in a fairly desperate situation where I've almost 5 weeks of historical video to download for a legal issue, from across 7 ring cameras.  I've got them set to maximum motion-sensitive, and longest recording time (2 minutes).  At this point there are thousands of videos (I'm guessing) and an urgent legal need to get them downloaded and to my lawyer for review before court.

I just found this site in my search for a better solution to downloading historical Ring videos than they provide.

I've started to look through the material here.  I've been searcjing for a high-enough-level description of what this is and possibly how to use/implement it.  I'm not yet even sure this project is the right place for me to begin.  But here goes....

I've a current CentOS 7.9 VM, I've downloaded the code from here and unpacked it; and am stuck.
I'm assuming I'll need Java, so I'll get that installed.  But I don't see sufficient details in the README and other files to help me bridge what I know (about Linux and scripting), to how to use/implement this.  Seems there are quite a few assumptions made that I'm not educated about.

I'm requesting help to get me started, please.

What I'd like to have sufficient assistance/direction (hand-holding---lol) to get to a point where using this API, in a linux VM, will give me the ability to provide it my RING account creds, select a camera by name (I have the info for each), iteratively select a day, and assuming the API can give me some feedback I can use that will 'list' the videos available for a given camera, for a given day.... I know how to script a bash wrapper around that to iterate through the list and (I HOPE) leverage this API again, to iterate through all the videos for a given day, download them, rename them to suit and go on to the next.  Then I can rinse/repeat to get everything down.  I can set things up to have a directory structure for each camera, sub-dir for each day within them.

Thank you.

**Comment by dgreif** [comment]
@Joe-Wulf sorry to hear you are going through a legal battle!  The main `ring-client-api` package is intended to be installed using `npm`, and executed using [NodeJS](https://nodejs.org/).  If you are not familiar with NodeJS and javascript (not Java), then the task you want to complete would definitely take some time.  Unfortunately it's not something I currently have the bandwidth to help out with.  As an alternative, I'd recommend logging into your account on ring.com, where you can easily view and download videos for specific cameras.   If that approach isn't sufficient, then I'd recommend reaching out to Ring support to see if they can send you a bulk export of your videos.

**Reply by dgreif** [reply]
@Joe-Wulf I went ahead and created an example project that demonstrates how to set up a Node.js project to interact with this package.  It's not tuned to downloading videos, but should give you a starting point if you really want to pursue the code route.  Best of luck! https://github.com/dgreif/ring-client-example

**Reply by Joe-Wulf** [reply]
@dgrief---Thank you, for all the effort you put into this for me.  I will begin to take a look at this this coming week as my own bandwidth allows.  Very much appreciated.
